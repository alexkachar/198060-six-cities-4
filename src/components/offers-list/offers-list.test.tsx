import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import OffersList from './offers-list';
import {BrowserRouter} from 'react-router-dom';

import mockOffers from '../../test-data/offers';
import Offer from '../../interfaces/offer';
import mockStore from '../../test-data/mock-store';

const OFFERS: Offer[] = mockOffers;

it(`OffersList renders correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={mockStore}>
          <BrowserRouter>
            <OffersList
              offers={OFFERS}
              cardType={`cities`} />
          </BrowserRouter>
        </Provider>, {
          createNodeMock: () => document.createElement(`div`)
        }
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
