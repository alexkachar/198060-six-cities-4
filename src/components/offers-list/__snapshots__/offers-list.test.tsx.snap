// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersList renders correctly 1`] = `
Array [
  <article
    className="place-card cities__place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €248
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button button"
          href="/"
          onClick={[Function]}
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating 
            4.4
          </span>
        </div>
      </div>
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <h2
          className="place-card__name"
        >
          Penthouse, 4-5 rooms + 5 balconies
        </h2>
      </a>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>,
  <article
    className="place-card cities__place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__image-wrapper cities__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €248
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button button"
          href="/"
          onClick={[Function]}
        >
          <svg
            className="place-card__bookmark-icon"
            height={19}
            width={18}
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating 
            4.4
          </span>
        </div>
      </div>
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <h2
          className="place-card__name"
        >
          Penthouse, 4-5 rooms + 5 balconies
        </h2>
      </a>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>,
]
`;
